foyer/forcefield.py

```diff
@@ -298,7 +345,26 @@ def createSystem(self, topology, atomtype=True, nonbondedMethod=NoCutoff,
             the newly created System
         """
         if atomtype:
-            find_atomtypes(topology, forcefield=self)
+            independent_residues = _check_independent_residues(topology)
```

Comment:
I like having atomtyping by residue act as the default behavior, but perhaps (as we spoke about regarding unit tests) having a `by_residue` argument to `createSystem()`, or something of that sort would be useful just so that we have the option to revert to the old behavior if desired.  We are already passing `**kwargs` to `createSystem()`, so we should be able to pass `by_residue` straight from `pmd.Structure.apply()` or `mb.Compound.apply()`.